src/include/catalog/gp_matview_aux.h

```diff
@@ -27,6 +27,7 @@ CATALOG(gp_matview_aux,7153,GpMatviewAuxId) BKI_SHARED_RELATION
 {
 	Oid			mvoid; 	/* materialized view oid */
 	NameData	mvname; /* materialized view name */
+	bool		has_foreign;	/* view query has foreign tables? */
```

Comment:
use `char relkind` has more extensibility.

Reply:
This is intended, for future view have multiple tables.
Think about that: `view mv1 as select * from t1 join foreign_table_t2 on true`;
If one of the tables is foreign table, the field `has_foreign` is true for mv1, and should avoid using REFRESH fast path.